core/src/main/java/org/testcontainers/containers/output/WaitingConsumer.java

```diff
@@ -73,14 +73,14 @@ public void waitUntil(Predicate<OutputFrame> predicate, int limit, TimeUnit limi
      */
     public void waitUntil(Predicate<OutputFrame> predicate, long limit, TimeUnit limitUnit, int times)
         throws TimeoutException {
-        long expiry = limitUnit.toMillis(limit) + System.currentTimeMillis();
+        long expiry = limitUnit.toNanos(limit) + System.nanoTime();
 
         waitUntil(predicate, expiry, times);
     }
 
     private void waitUntil(Predicate<OutputFrame> predicate, long expiry, int times) throws TimeoutException {
         int numberOfMatches = 0;
-        while (System.currentTimeMillis() < expiry) {
+        while (System.nanoTime() < expiry) {
```

Comment:
nanoTime is subject to overflow/wrapping and the addition of  an offset may have wrapped.


Whilst this could have happened with mills you wouldn't care for a few more years.  nanoTime is relative to a random point in time so it could be a very large number or a very small number, positive or megative

Reply:
Fixed